bindings/rust/s2n-tls/src/raw/connection.rs

```diff
@@ -362,10 +362,37 @@ impl Connection {
     pub fn negotiate(&mut self) -> Poll<Result<&mut Self, Error>> {
         let mut blocked = s2n_blocked_status::NOT_BLOCKED;
 
-        match unsafe { s2n_negotiate(self.connection.as_ptr(), &mut blocked).into_result() } {
-            Ok(_) => Ok(self).into(),
-            Err(err) if err.is_retryable() => Poll::Pending,
-            Err(err) => Err(err).into(),
+        unsafe {
+            s2n_negotiate(self.connection.as_ptr(), &mut blocked)
+                .into_poll()
+                .map_ok(|_| self)
+        }
+    }
+
+    pub fn send(&mut self, buf: &[u8]) -> Poll<Result<usize, Error>> {
```

Comment:
can you add doc comments for the new methods?

Reply:
Good call. While writing the shutdown documentation, I realized I wasn't shutting down the inner stream in TlsStream 🤦